python/ray/data/_internal/execution/interfaces/physical_operator.py

```diff
@@ -542,14 +542,25 @@ def pending_processor_usage(self) -> ExecutionResources:
         """
         return ExecutionResources(0, 0, 0)
 
-    def base_resource_usage(self) -> ExecutionResources:
-        """Returns the minimum amount of resources required for execution.
+    def min_resource_usage(self) -> ExecutionResources:
+        """Returns the minimum resources to start the operator and make progress.
 
         For example, an operator that creates an actor pool requiring 8 GPUs could
-        return ExecutionResources(gpu=8) as its base usage.
+        return ExecutionResources(gpu=8) as its minimum usage.
+
+        This method is used by the resource manager to reserve minimum resources for
+        the operator.
         """
         return ExecutionResources()
 
+    def max_resource_usage(self) -> ExecutionResources:
+        """Returns the maximum resources that this operator can use right now.
+
+        This method is used by the resource manager to ensure that it doesn't
+        over-provision resources for this operator.
+        """
+        return ExecutionResources.for_limits()
```

Comment:
`*_resource_usage` makes it look like it's referring to an actual usage.

I'd suggest to:

 - Combine into 1 method
 - Name it `min_max_usage_bounds`